creator-node/src/snapbackSM/syncHistoryAggregator.js

```diff
@@ -62,45 +81,89 @@ class SyncHistoryAggregator {
     }
   }
 
+  /**
+   * Retrieves the current key's ttl. If the key doesn't exist or is indefinite,
+   * return the default expiration time of 7 days.
+   * @param {String} key key to retrieve from redis
+   * @returns expiration time in seconds
+   */
   static async getKeyTTL (key) {
-    const ttl = await redisClient.ttl(key)
-    return ttl && ttl > 0 ? ttl : EXPIRATION
+    try {
+      const ttl = await redisClient.ttl(key)
+      return ttl && ttl > 0 ? ttl : EXPIRATION
+    } catch (e) {
+      return EXPIRATION
+    }
   }
 
-  static async getAggregateSyncData () {
-    const { success, fail } = SyncHistoryAggregator.getAggregateSyncKeys()
+  /**
+   * Returns the aggregate sync data of the current day's number of successful, failed,
+   * and triggered syncs
+   * @param {Object} logContext the log context off of the Express req object
+   * @returns an object of the current day's aggregate sync count like
+   *    {triggered: <number>, success: <number>, fail: <number>}
+   */
+  static async getAggregateSyncData (logContext) {
+    const logger = genericLogger.child(logContext)
+    let currentAggregateData = {
+      success: 0,
+      fail: 0,
+      triggered: 0
```

Comment:
just a thought, how would you feel about defaulting these to -1, so we can differentiate between a node that has actually 0 syncs vs a node that failed to fetch the sync data

Reply:
I do not think this is a good idea. setting 0 for each of these states means what it means -- that 0 syncs have occurred. if a sync were to ever fail and the tracking logic is successful, the states will increment as they should. setting these states with an initial value of -1 only adds complexity, as now we have to add additional logic to handle the very first increment on tracking syncs.

Reply:
if the node failed to retrieve data, the `fail` key would increment by 1

Reply:
agreed it adds more logic - its potentially useful but probably not worth. down to just leave as is